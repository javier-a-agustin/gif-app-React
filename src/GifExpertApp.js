import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Adventure']);

    return(
        <>
            <h2>Gifs</h2>

            <AddCategory setCategories={ setCategories }/>
            <hr></hr>
           

            <ul>

                {
                    categories.map( category => (
                        <GifGrid key={ category } category={ category }/>
                    ))
                }

            </ul>
        </>
    );
};

export default GifExpertApp;