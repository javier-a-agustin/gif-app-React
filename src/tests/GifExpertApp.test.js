import React from 'react';
import '@testing-library/jest-dom';
import GifExpertApp from '../GifExpertApp';
import { shallow } from 'enzyme';

describe('Pruebas en GifExpertApp', () => {

    test('debe mostrar el componente correctamente', () => {
        const wrapper = shallow(<GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe mostrar una lista de categoras', () => {
        const categories = ['Adventure', 'Transformers'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} /> );
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(categories.length);
    })

});