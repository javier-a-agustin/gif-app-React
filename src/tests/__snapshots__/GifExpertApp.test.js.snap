// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    Gifs
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;

exports[`Pruebas en GifExpertApp debe mostrar una lista de categoras 1`] = `
<Fragment>
  <h2>
    Gifs
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="Adventure"
      key="Adventure"
    />
    <GifGrid
      category="Transformers"
      key="Transformers"
    />
  </ul>
</Fragment>
`;
